File: problems/hard/bear_and_gene/__tests__/test_bear_and_gene.js
const steadyGene = require('../bear_and_gene');

function testSteadyGene() {
    const testCases = [
        {
            input: 'GAAATAAA',
            expected: 5,
            description: 'Example 1',
        },
        {
            input: 'GACT',
            expected: 0,
            description: 'Already steady gene',
        },
        {
            input: 'AAAACCCCGGGGTTTT',
            expected: 0,
            description: 'Large balanced gene',
        },
        {
            input: 'AAGTGCCT',
            expected: 0,
            description: 'Another case',
        },
        {
            input: 'AAAAACCCGGTT',
            expected: 2,
            description: 'Edge case',
        }
    ];

    testCases.forEach(testCase => {
        const { input, expected, description } = testCase;
        const result = steadyGene(input);

        if (result === expected) {
            console.log(`✅ Passed: ${description}`);
        } else {
            console.log(`❌ Failed: ${description}`);
            console.log(`  Expected: ${expected}`);
            console.log(`  Got: ${result}`);
        }
    });
}

testSteadyGene();

File: problems/hard/bear_and_gene/__tests__/TestBearAndGene.java
package problems.hard.bear_and_gene.__tests__;

import problems.hard.bear_and_gene.BearAndGene;

public class TestBearAndGene {
    public static void main(String[] args) {
        test("GAAATAAA", 5);
        test("GACT", 0);
        test("AAAACCCCGGGGTTTT", 0);
        test("AAGTGCCT", 0);
        test("AAAAACCCGGTT", 2);
    }

    private static void test(String gene, int expected) {
        int result = BearAndGene.steadyGene(gene);
        if (result == expected) {
            System.out.println("✅ Passed for input: " + gene);
        } else {
            System.out.println("❌ Failed for input: " + gene + ". Expected: " + expected + ", got: " + result);
        }
    }
}
